examples/onnxruntime/inceptionV3/inceptionv3.py

```diff
@@ -0,0 +1,276 @@
+              str(img_count) + "Images from dataset")
+        print(bad_image + "Images caused RunTimeErrors during preprocessing")
+        if verbose:
+            print("List of bad Images:\n" + str(bad_img_list))
+
+    return input_batch
+
+
+def main():
+    flags = parse_input_args()
+
+    if flags.verbose:
+        print(flags)
+
+    if flags.verbose:
+        print("Reading in Imagenet classes")
+
+    # Read the categories
+    with open("../dataset/imagenet_classes.txt", "r") as f:
+        categories = [s.strip() for s in f.readlines()]
+
+    if flags.verbose:
+        print("Getting Exported Model from Torch")
+
+    # Export the model to ONNX
+    image_height = 299
+    image_width = 299
+    x = torch.randn(flags.batch,
+                    3,
+                    image_height,
+                    image_width,
+                    requires_grad=True)
+    inception_v3(x)
+    torch.onnx.export(
+        inception_v3,  # model being run
+        x,  # model input (or a tuple for multiple inputs)
+        "inception_v3.onnx",  # where to save the model (can be a file or file-like object)
+        export_params=
+        True,  # store the trained parameter weights inside the model file
+        opset_version=14,  # the ONNX version to export the model to
+        do_constant_folding=
+        True,  # whether to execute constant folding for optimization
+        input_names=['input'],  # the model's input names
+        output_names=['output'],  # the model's output names
+        verbose=flags.verbose)
+
+    # Quantize the model
+    if flags.fp16:
+        if flags.verbose:
+            print("FP16 Quantization Enabled")
+        os.environ["ORT_MIGRAPHX_FP16_ENABLE"] = "1"  # Enable FP16 precision
+    else:
+        os.environ["ORT_MIGRAPHX_FP16_ENABLE"] = "0"  # Disable FP32 precision
+
+    session_ops = onnxruntime.SessionOptions()
+    if flags.verbose:
+        session_ops.log_verbosity_level = 0
+        session_ops.log_severity_level = 0
+
+    session_fp32 = onnxruntime.InferenceSession(
+        "inception_v3.onnx",
+        providers=['MIGraphXExecutionProvider'],
+        sess_options=session_ops)
+
+    if flags.verbose:
+        print("Preprocessing Batched Images")
+
+    # Preproccess and batch images
+    input_batch = preprocess_images(flags.image_dir, image_height, image_width,
+                                    flags.batch, flags.verbose)
+
+    if flags.verbose:
+        print("Running samples")
+
+    latency = []
+    for i in range(flags.run):
+        run_sample(session_fp32, categories, latency, input_batch, flags.top,
+                   flags.batch)
+
+    if flags.verbose:
+        print("Running Complete")
+        print(latency)
+
+    if flags.QPS:
+        print("inception_v3, Rate = {} QPS".format(
+            format((((flags.batch)) / (sum(latency) / len(latency))), '.2f')))
+    else:
+        print("inception_v3, Total time = {} ms".format(
```

Comment:
Call this "Average execution time" rather than "Total time", and in the calculation for both qps and time remove the first (cold start) measurement. You can simply replace `latency` with `latency[1:]`

Reply:
Total time is to Mirror what we do with our driver but sure I can change it. I think there was a comment about setting it to Total time instead of just "latency". Average is better here though

Reply:
Done